hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/SnapshotLoadQuerySplitter.java

```diff
@@ -0,0 +1,77 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.hudi.utilities.sources.helpers;
+
+import org.apache.hudi.common.config.TypedProperties;
+import org.apache.hudi.common.util.Option;
+import org.apache.spark.sql.Dataset;
+import org.apache.spark.sql.Row;
+
+/**
+ * Abstract splitter responsible for managing the snapshot load query operations.
+ */
+public abstract class SnapshotLoadQuerySplitter {
```

Comment:
1. This is not a helper. Move the abstraction out of `helpers` package.
2. How about keeping the base class name simply `IncrementalLoadSplitter` as it is used only for incremental source?
3. Why are we not providing any concrete out-of-box implementation?
4. Can we add unit test for the splitter `getNextCheckpoint`?

Reply:
Moved to sources package . 
Name should indicate it's only for snapshot load query of incremental sources , so kept as it is . 
Added unit test for with test implementation , concrete implementation at this point .